packages/react-native-codegen/src/parsers/typescript/modules/index.js

```diff
@@ -76,6 +152,38 @@ function translateTypeAnnotation(
     resolveTypeAnnotation(typeScriptTypeAnnotation, types);
 
   switch (typeAnnotation.type) {
+    case 'TSArrayType': {
+      return translateArrayTypeAnnotation(
+        hasteModuleName,
+        types,
+        aliasMap,
+        cxxOnly,
+        'Array',
+        typeAnnotation.elementType,
+        nullable,
+      );
+    }
+    case 'TSTypeOperator': {
+      if (
+        typeAnnotation.operator === 'readonly' &&
+        typeAnnotation.typeAnnotation.type === 'TSArrayType'
+      ) {
+        return translateArrayTypeAnnotation(
+          hasteModuleName,
+          types,
+          aliasMap,
+          cxxOnly,
+          'ReadonlyArray',
+          typeAnnotation.typeAnnotation.elementType,
```

Comment:
Is this correct? I can can see in the previous case that the line was just `typeAnnotation.elementType` instead of `typeAnnotation.typeAnnotation.elementType`

Reply:
This is intended.
For `T[]`, it is a `TSArrayType` on a `TSTypeReference`.
For `readonly T[]`, it is a `TSTypeOperator` on a `TSArrayType` on a `TSTypeReference`.

This is why it has one more `.typeAnnotation`.